### PR TITLE
[SPARK-7504] [YARN] NullPointerException when initializing SparkContext…

### PR DESCRIPTION
**Pull request by zzvara**
… in YARN-cluster mode

Added check if ApplicationMaster has been initialized with additional error message.

### CODE REVIEW

**Comment by andrewor14** [comment]
ok to test

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 1]
```diff
@@ -573,10 +573,19 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if(master == null) {
```
Comment (srowen) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: (Should be a space after `if` but that can be fixed on merge; might fix that if you make any other change.) 

I think the text is OK although I suppose that as of 1.4 there's the launcher API, but, probably not worth writing a man page here.

Comment (ehnalis) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: I'll just let you fix it on merge.

Also, thank you for attracting my attention to the launcher API.

**Comment by andrewor14** [comment]
I think @vanzin left some comments on the JIRA that should be addressed here.

**Comment by ehnalis** [comment]
@andrewor14 Can I use this PR for two issues?

**Comment by srowen** [comment]
Eh, this now seems to have some unrelated changes. These are quite unrelated JIRAs, so you certainly shouldn't make one PR for both. 

I think SPARK-7504 should just be closed, along with this PR, and a new one can be opened for your other separate changes. I am not sure 7504 is worth bothering making a new exception for.

**Pull request closed without merging**